### PR TITLE
Metadata class updates.

### PR DESCRIPTION
**Pull request by pshriwise**
## Description
Updates to the `dagmcMetaData` class. 

This addresses a comment from @bam241 in #688 about some potential improvements to style in the `dagmcMetaData` class.

It addresses an output type in the UWUW class as well.

### CODE REVIEW

**Comment by gonuke** [comment]
Thanks @pshriwise.  

I seem to remember a discussion about the pros/cons of using `auto`, but can't remember the outcome.  Have we already committed to that path?

**Comment by pshriwise** [comment]
My opinion is that it's handy to use in the right situations, but should be used cautiously. In certain situations it makes the code less verbose and more readable. Here I felt using it to replace the very long type definitions was worth it. I do try to avoid using it in places where the compiler will infer a type as that can lead to unexpected problems however. Using it as the return type in function signatures is one such example.

IIRC @makeclean may have strong opinions about `auto`. I'm happy to defer to any strong objections about its use.

**Comment by makeclean** [comment]
My only objection is the obfuscation it can cause with multi-header types, but that lack of clarity is fixed by an ide with type discovery. So I don't know that I'm opposed totally, it's very handy but if youre not using an editor with type discovery then complex types will be painful. Also happy to defer
________________________________

**Comment by gonuke** [comment]
Apparently this failed the `astyle` check, although not reported here.

**Comment by pshriwise** [comment]
I applied the style guide command, but it made some changes there weren't necessarily.... stylish. I'm curious to see what CI does with it.

**Comment by gonuke** [comment]
This one passed CI (but I'm still not sure why it's not showing the results of tests here...) @bam241 ?

File: src/dagmc/dagmcmetadata.hpp [hunk 1]
```diff
@@ -47,13 +58,20 @@ class dagmcMetaData {
...
   void finalise_counters();
 
-  std::map<moab::EntityHandle, std::vector<std::string> > get_property_assignments(std::string property,
-      int dimension,
-      std::string delimiters,
-      bool remove_duplicates = true);
+  // Parse property for entities with the specified dimension and delimiters.
+  // Optionally remove duplicate property values if necessary.
+  std::map<moab::EntityHandle, std::vector<std::string>>
+                                                      get_property_assignments(std::string property,
```
Comment (pshriwise) [inline src/dagmc/dagmcmetadata.hpp:64]: Not sure why the style guide considers this to be better...

Comment (gonuke) [reply src/dagmc/dagmcmetadata.hpp:64]: agreed!  @bam241 has suggested that we switch to clang-format.... maybe soon...

**Comment by bam241** [comment]
> This one passed CI (but I'm still not sure why it's not showing the results of tests here...) @bam241 ?

there is a failure somewhere in the link between Travis-CI and GitHub, restarting the tests corresponding the the PR on CI ends usually does the trick :)

**Comment by pshriwise** [comment]
@makeclean agreed. We probably don't want to go crazy with it. Using it for native types like `double`, `int`, and even `std::vector` probably isn't worth it. It can really help with these nested template types though I think. If you're ok with using it sparingly I'll leave these changes as they are. 

Seems as though CI has reconnected here too, so that's good.

**Comment by gonuke** [comment]
Thanks @pshriwise - I'll merge this and we'll officially have `auto` in core DAGMC code....

**Pull request merged**